### PR TITLE
Use Qiskit Runtime `SamplerV2` in local mode instead of `BackendSamplerV2` in test

### PR DESCRIPTION
**Pull request by garrison**
This is a draft of something we could do once the next `qiskit-ibm-runtime` release comes out (we must wait until then because we need https://github.com/Qiskit/qiskit-ibm-runtime/pull/1558).  Once that happens, I will ~~likely further iterate on this~~ open another pull request that builds on this to fix some or (ideally) all of the following issues:
- #511
- #509 
- #505 
- #535

More info on this migration is at https://docs.quantum.ibm.com/api/migration-guides/local-simulators.

### CODE REVIEW

**Comment by garrison** [comment]
I'm going to add this to the 0.7.0 milestone because it'd be really nice to knock these items off and move to using fake backends in the docs as soon as possible.  Even if we were to merge this minimal PR before our next release, we could backport the other changes because ckt would already depend on the requisite qiskit-ibm-runtime version.  To enable this, though, we'll need a qiskit-ibm-runtime release, either 0.22.1 from the current stable branch or [0.23.0](https://github.com/Qiskit/qiskit-ibm-runtime/milestone/29).

**Comment by caleb-johnson** [comment]
Could you please remind me why we need to wait for another release of `qiskit-ibm-runtime` to knock out all of the issues in the original PR? Is that a dated comment?

File: test/cutting/test_cutting_workflows.py [hunk 1]
```diff
@@ -213,11 +203,9 @@ def test_reconstruction_with_samplerv2():
         num_samples=100,
     )
 
-    # Use BackendSamplerV2 with AerSimulator, following
-    # https://github.com/Qiskit/qiskit-aer/issues/2078#issuecomment-1971498534
+    # Use SamplerV2 in local mode with AerSimulator
     samplers = {
-        label: BackendSamplerV2(backend=AerSimulator())
-        for label in subexperiments.keys()
+        label: SamplerV2(backend=AerSimulator()) for label in subexperiments.keys()
```
Comment (caleb-johnson) [inline test/cutting/test_cutting_workflows.py:208]: Any particular reason for choosing `AerSimulator` over [GenericBackend](https://docs.quantum.ibm.com/api/qiskit/qiskit.providers.fake_provider.GenericBackendV2)?

**Comment by garrison** [comment]
> Could you please remind me why we need to wait for another release of `qiskit-ibm-runtime` to knock out all of the issues in the original PR? Is that a dated comment?

We need to bump the version of qiskit-ibm-runtime even for this PR to work because it is the first version that includes https://github.com/Qiskit/qiskit-ibm-runtime/pull/1558.

I bumped the version of Aer here, too, so that we can backport #522 (if it doesn't land in time for the release) without needing to further bump any versions on the stable branch.

File: test/cutting/test_cutting_workflows.py (hunk 1 above)
Comment (garrison) [reply test/cutting/test_cutting_workflows.py:208]: I did not consider it; I was just following https://docs.quantum.ibm.com/api/migration-guides/local-simulators#aersimulator, which uses `AerSimulator` as a backend.

Comment (caleb-johnson) [reply test/cutting/test_cutting_workflows.py:208]: OK thanks, it looks like it's just strictly more performant than the statevector stuff in Qiskit

**Review by caleb-johnson — APPROVED** [review]
This looks great, thanks!

Line 51 in the test uses `GenericBackendV2`. Should we make a test explicitly testing against that backend and use `AerSimulator` everywhere by default? The only places we would need to switch would be here and Tutorial 2, cutting for depth.

**Comment by garrison** [comment]
> Line 51 in the test uses `GenericBackendV2`. Should we make a test explicitly testing against that backend and use `AerSimulator` everywhere by default? The only places we would need to switch would be here and Tutorial 2, cutting for depth.

Actually, in #522 I am transitioning to using Qiskit Runtime in local mode with fake backends in all the tutorials, so I think that, the test you mention, plus `AerSimulator` in this PR's test covers all our bases.

**Pull request merged**